### ISSUE TITLE
CSV files detect as reject file

### ISSUE THREAD

**Issue by indraakkk** [fortana-co/react-dropzone-uploader#11]
I was able to check is the image file or not, but when accepted file is CSV, `extra.reject = true` while file dragging

**How to reproduce error**
```
<DropZone
   accept = ".csv"
   inputContent={(files, extra) => ( console.log(extra) ) }
/>
```

drag your .csv file to dropzone area, then check your console log, you can see `reject = true`, but after you release mouse click, file became accepted

Please help to assist

**Comment by kylebebak** [comment]
Try changing your code to this:

```js
<DropZone
   accept = "text/csv"
   inputContent={(files, extra) => ( console.log(extra) ) }
/>
```

Now your `.csv` file won't be rejected on drag over, because we're using a MIME type instead of a file name. The is because the file name, unlike the MIME type, isn't available in the drag handler, but only on the drop handler. This is a limitation of `event.dataTransfer`.

I'm not sure what the best way to handle this is. There are two options:

1. Reject files in the drag event that have no file name if the accept attribute matches them by file name (what we're doing currently)
2. Accept these files in the drag event

I think for the time being the best solution is to use a MIME type in the accept attribute, like `text/csv`, because this will work for both the drag and drop events.

**Comment by kylebebak** [comment]
I've given this some thinking. The file name is available in the drop event **but not the drag event**. This means if you use an accept attribute that matches file name, like `.csv`, instead of MIME type, like `text/csv`, any implementation is bound to result in one of two inconsistencies. Imagine you use `accept=".csv"`.

1. If we reject files with no file name, files like "file.csv" will be rejected erroneously on drag.
2. If we accept files with no name, files like "file.xml" will be accepted erroneously on drag.

Currently we're doing option 1. Which is preferable?

I'm inclined to think that option 2 is preferable, because once the user drops "file.xml", the user gets immediate feedback because the file is rejected. This probably won't seem like a bug from the user's POV.

With option 1, you have situations where the dropzone turns red on drag, and then the file is accepted on drop, which seems like a bug from the user's POV.

**Comment by kylebebak** [comment]
This new behavior just shipped in `2.9.0`.

@indraakkk 
Closing this. Give it a try, and let me know if you have any trouble. Thanks for creating the issue!
